ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageServiceFactory.java

```diff
@@ -51,11 +49,11 @@
    * @param responseHandler the {@link RestResponseHandler} that can be used to submit responses that need to be sent
    *                        out.
    * @param router the {@link Router} to use.
-   * @param notifier the {@link Notifier} to use.
+   * @param accountService the {@link AccountService} to use.
    * @throws IllegalArgumentException if any of the arguments are null.
    */
   public AmbryBlobStorageServiceFactory(VerifiableProperties verifiableProperties, ClusterMap clusterMap,
-      RestResponseHandler responseHandler, Router router, Notifier notifier) {
+      RestResponseHandler responseHandler, Router router, AccountService accountService) {
     if (verifiableProperties == null || clusterMap == null || responseHandler == null || router == null) {
       throw new IllegalArgumentException("Null arguments were provided during instantiation!");
```

Comment:
would be better if we check `accountService == null` because `accountService` is directly passed to `getBlobStorageService` method.

Reply:
Also added a test